CMakeLists.txt

```diff
@@ -0,0 +1,254 @@
+cmake_minimum_required(VERSION 3.5)
+
+project(OpenTTD)
+
+if (CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
+    message(FATAL_ERROR "In-source builds not allowed. Please run \"cmake ..\" from the bin directory")
+endif (CMAKE_SOURCE_DIR STREQUAL CMAKE_BINARY_DIR)
```

Comment:
I probably have set this before, but just remove the arguments to all end*() calls. It makes code much easier to maintain.